tests/ProxyManagerTest/Exception/FileNotWritableExceptionTest.php

```diff
@@ -15,25 +16,16 @@
  */
 final class FileNotWritableExceptionTest extends TestCase
 {
-    public function testFromInvalidMoveOperation() : void
+    public function testFromPrevious() : void
     {
-        $exception = FileNotWritableException::fromInvalidMoveOperation('/tmp/a', '/tmp/b');
+        $previous = $this->getMockBuilder(FileWriterException::class)
```

Comment:
I'd probably instantiate a real `FileWriterException` here, instead of a mock. Psalm chokes here because of the fluent interface usage, but getting rid of the mock directly would simplify things enough.

Reply:
There are 4 different exceptions which implements the same interface:
https://github.com/webimpress/safe-writer/tree/master/src/Exception
All these exceptions have private constructor, and all public static methods are marked as internal.
I believe mocking in that place is the most appropriate, and we should skip somehow psalm error here.

Reply:
Dropped fluent interface usage and it works now ! :) Thanks, @Ocramius!

Reply:
Nice! BTW, since we're mocking only the interface, the `final public function getMessage()` isn't really enforced anyway (in theory)